llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -27519,6 +27534,32 @@ static SDValue performMULLCombine(SDNode *N,
   return SDValue();
 }
 
+static SDValue performPTestFirstCombine(SDNode *N,
+                                        TargetLowering::DAGCombinerInfo &DCI,
+                                        SelectionDAG &DAG) {
+  if (DCI.isBeforeLegalize())
+    return SDValue();
+
+  SDLoc DL(N);
+  auto Mask = N->getOperand(0);
+  auto Pred = N->getOperand(1);
+
+  if (Pred->getOpcode() == AArch64ISD::REINTERPRET_CAST)
+    Pred = Pred->getOperand(0);
+
+  if (!isLane0KnownActive(Mask))
+    return SDValue();
```

Comment:
Perhaps move this before looking through Pred's reinterpret_cast because this is what makes that code logically safe.